engine/src/org/pentaho/di/www/WebServer.java

```diff
@@ -77,9 +78,11 @@
 
   private String passwordFile;
 
+  private SslConfiguration sslConfig;
+
   public WebServer( LogChannelInterface log, TransformationMap transformationMap, JobMap jobMap,
       SocketRepository socketRepository, List<SlaveServerDetection> detections, String hostname, int port,
-      boolean join, String passwordFile ) throws Exception {
+      boolean join, String passwordFile, SslConfiguration sslConfig ) throws Exception {
```

Comment:
As a public "API" (constructor), should we keep the original as deprecated and have it call a new constructor with a null SSL config?